lib/galaxy/webapps/galaxy/api/uploads.py

```diff
@@ -0,0 +1,42 @@
+"""
+API operations for uploaded files in storage.
+"""
+import logging
+import os
+
+from galaxy import exceptions
+from galaxy.web import expose_api_anonymous
+from galaxy.web.base.controller import BaseAPIController
+
+log = logging.getLogger(__name__)
+
+
+class UploadsAPIController(BaseAPIController):
+
+    @expose_api_anonymous
+    def index(self, trans, **kwd):
+        raise exceptions.NotImplemented("Listing uploads is not implemented.")
+
+    @expose_api_anonymous
+    def create(self, trans, payload, **kwd):
+        """
+        POST /api/uploads/
+        """
+        session_id = payload.get("session_id")
+        session_start = payload.get("session_start")
+        if session_id is None:
+            raise exceptions.MessageException("Requires a session id.")
+        if session_start is None:
+            raise exceptions.MessageException("Requires a session start.")
+        target_file = "%s/%s" % (trans.app.config.new_file_path, session_id)
+        target_size = 0
+        if os.path.exists(target_file):
+            target_size = os.path.getsize(target_file)
+        if session_start != target_size:
+            raise exceptions.MessageException("Incorrect session start.")
+        source = payload.get("session_chunk")
+        with open(target_file, "a") as f:
+            f.write(source.file.read())
```

Comment:
Should this be read in chunks? I suppose this can fill up the memory.

Reply:
This is already the file_chunk. I added additional checks to verify the chunk size.

Reply:
Depending on the configured chunk size, this can still be quite big. You can replace this line with something like:
```python
read_chunk_size = 2 ** 16
while True:
    read_chunk = source.file.read(read_chunk_size)
    if not read_chunk:
        break
    f.write(read_chunk)
```

Reply:
I see, ok cool.